### ISSUE TITLE
WebUI becomes unresponsive

### ISSUE THREAD

**Issue by slim186** [HaveAGitGat/Tdarr#386]
**Describe the bug**
Occasionally, the WebUI module will become unresponsive.  

**To Reproduce**
Seems more likely to happen when large numbers of files are being processed quickly (after a Scan Fresh on a finished library for example).

**Expected behavior**
WebUI continues to respond normally

**Please provide the following information:**
- Config files  
Tdarr_WebUI_Config.json:
{
  "webUIPort": 8264,
  "serverIP": "0.0.0.0",
  "serverPort": 8266
}

- Log files  
systemd/tdarr-webui:
Feb 24 09:02:22 odin systemd[1]: Started Tdarr WebUI.
Feb 24 09:02:24 odin Tdarr_WebUI[22216]: undefined
Feb 24 09:02:26 odin Tdarr_WebUI[22216]: /opt/Tdarr2/configs/Tdarr_WebUI_Config.json
Feb 24 09:02:26 odin Tdarr_WebUI[22216]: { webUIPort: 8264, serverIP: '0.0.0.0', serverPort: 8266 }
Feb 24 09:02:27 odin Tdarr_WebUI[22216]: [HPM] Proxy created: /  -> http://0.0.0.0:8266
Feb 24 09:02:28 odin Tdarr_WebUI[22216]: [2021-02-24T09:02:28.163] [INFO] Tdarr_WebUI - Tdarr_Server is alive at 0.0.0.0:8266
Feb 24 09:02:28 odin Tdarr_WebUI[22216]: Tdarr_WebUI running at http://localhost:8264
Feb 24 09:02:28 odin Tdarr_WebUI[22216]: [HPM] Error occurred while trying to proxy request /socket.io/?EIO=4&transport=polling&t=NVKa1S6&sid=duJnx6KM5a0yi3clAAAr from 192.168.5.40:8264
-- manually restarted tdarr-webui here --
Feb 24 09:28:17 odin systemd[1]: Stopping Tdarr WebUI...
Feb 24 09:28:19 odin systemd[1]: tdarr-webui.service: Main process exited, code=killed, status=15/TERM
Feb 24 09:28:19 odin systemd[1]: tdarr-webui.service: Succeeded.
Feb 24 09:28:19 odin systemd[1]: Stopped Tdarr WebUI.
Feb 24 09:28:19 odin systemd[1]: Started Tdarr WebUI.
Feb 24 09:28:19 odin Tdarr_WebUI[13875]: undefined
Feb 24 09:28:20 odin Tdarr_WebUI[13875]: /opt/Tdarr2/configs/Tdarr_WebUI_Config.json
Feb 24 09:28:20 odin Tdarr_WebUI[13875]: { webUIPort: 8264, serverIP: '0.0.0.0', serverPort: 8266 }
Feb 24 09:28:20 odin Tdarr_WebUI[13875]: [HPM] Proxy created: /  -> http://0.0.0.0:8266
Feb 24 09:28:20 odin Tdarr_WebUI[13875]: [HPM] Proxy created: /  -> http://0.0.0.0:8266
Feb 24 09:28:20 odin Tdarr_WebUI[13875]: [2021-02-24T09:28:20.575] [INFO] Tdarr_WebUI - Tdarr_Server is alive at 0.0.0.0:8266
Feb 24 09:28:20 odin Tdarr_WebUI[13875]: Tdarr_WebUI running at http://localhost:8264

 - OS: Debian Buster (non-Docker)
 - Browser Firefox
 - Version 2.00.06

**Comment by jaycedk** [comment]
I have seen this on windows 10 as well.
Also with Firefox, dont know if firefox is the issue.
Will test with chrome.

**Comment by jaycedk** [comment]
OK been testing in chrome, and that works.
Its been running fine for 3 hours, where in Firefox 10 - 15 min and webui stopped responding.
Guess its a oddness in firefox.

**Comment by jaycedk** [comment]
@slim186 can you try in chrome ?

**Comment by jcouch93** [comment]
@jaycedk I can confirm that Chrome works when FF has the page locked up.

**Comment by tyhawk** [comment]
I have the same issue on Firefox, running on Linux with a Tdarr-Server running in a docker container (official build).
I have to admit that I've not seen this issue since I am running 2.00.07, but I will report it here if I run into it again.

**Comment by tyhawk** [comment]
Just run into it. Seeing this in the logs:
```
[2021-03-22T16:05:45.533] [INFO] Tdarr_Server - Online,
[2021-03-22T16:05:45.647] [INFO] Tdarr_Server - 15 files sending,
[2021-03-22T16:05:45.649] [INFO] Tdarr_Server - [0.001s] 15 files sent,
[2021-03-22T16:08:07.964] [INFO] Tdarr_Server - Finished in 142.432s,
[2021-03-22T16:08:07.977] [INFO] Tdarr_Server - Scanner 3oj-Hi9Lw6:Finished,
[2021-03-22T16:08:07.980] [WARN] Tdarr_Server - Exit approved. Closing process.,
[2021-03-22T16:08:09.008] [INFO] Tdarr_Server - 3oj-Hi9Lw6 File scanner exited,
[2021-03-22T16:28:31.628] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e01 - The Original.mkv,
[2021-03-22T16:28:31.629] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e02 - Chestnut.mkv,
[2021-03-22T16:28:31.630] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e03 - The Stray.mkv,
[2021-03-22T16:28:31.631] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e04 - Dissonance Theory.mkv,
[2021-03-22T16:28:31.631] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e05 - Contrapasso.mkv,
[2021-03-22T16:28:31.632] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e06 - The Adversary.mkv,
[2021-03-22T16:28:31.633] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e07 - Trompe LOeil.mkv,
[2021-03-22T16:28:31.634] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e08 - Trace Decay.mkv,
[2021-03-22T16:28:31.634] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e09 - The Well-Tempered Clavier.mkv,
[2021-03-22T16:28:31.634] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e10 - The Bicameral Mind.mkv,
[2021-03-22T16:28:31.645] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e05 - Caravan.mkv,
[2021-03-22T16:28:31.646] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e06 - Its all in the game.mkv,
[2021-03-22T16:28:31.647] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e07 - Groot nieuws.mkv,
[2021-03-22T16:28:31.647] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e08 - Schijn bedriegt.mkv,
[2021-03-22T16:28:31.648] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e09 - De vrijgezellen.mkv,
[2021-03-22T16:28:31.650] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e10 - Onverwachte wending.mkv,
[2021-03-22T16:28:31.655] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e11 - Omkoopschandaal.mkv,
[2021-03-22T16:28:31.656] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e12 - Race tegen de klok.mkv,
[2021-03-22T16:28:31.657] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e13 - Moederziel en kinderleed.mkv,
[2021-03-22T16:28:31.680] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 2/Westworld - s02e02 - Reunion.mkv,
[2021-03-22T16:28:31.681] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 2/Westworld - s02e03 - Virtù e Fortuna.mkv,
[2021-03-22T16:28:31.681] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 2/Westworld - s02e05 - Akane No Mai.mkv,
[2021-03-22T16:28:31.682] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Westworld/Season 2/Westworld - s02e07 - Les Écorchés.mkv,
[2021-03-22T16:28:31.723] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e01 - Nieuwbloed.mkv,
[2021-03-22T16:28:31.723] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e02 - Het drama.mkv,
[2021-03-22T16:28:31.724] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e03 - Oude bekende.mkv,
[2021-03-22T16:28:31.725] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e04 - Russische roulette.mkv,
[2021-03-22T16:28:31.725] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e05 - Vader.mkv,
[2021-03-22T16:28:31.726] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e06 - Uitweg.mkv,
[2021-03-22T16:28:31.726] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e07 - Geld.mkv,
[2021-03-22T16:28:31.726] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e08 - Creatief denkien.mkv,
[2021-03-22T16:28:31.729] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e09 - Ontvoerd.mkv,
[2021-03-22T16:28:31.730] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e10 - Tante Madonna.mkv,
[2021-03-22T16:28:31.730] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e11 - Voorzorgen.mkv,
[2021-03-22T16:28:31.731] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e12 - Oost west, thuis best.mkv,
[2021-03-22T16:28:31.732] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e13 - De bedreigde bruid.mkv,
[2021-03-22T16:29:04.277] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e01 - The Original.mp4,
[2021-03-22T16:29:04.281] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e03 - The Stray.mp4,
[2021-03-22T16:29:04.289] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e05 - Contrapasso.mp4,
[2021-03-22T16:29:04.291] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e04 - Dissonance Theory.mp4,
[2021-03-22T16:29:04.293] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e06 - The Adversary.mp4,
[2021-03-22T16:29:04.294] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e07 - Trompe LOeil.mp4,
[2021-03-22T16:29:04.296] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e02 - Chestnut.mp4,
[2021-03-22T16:29:04.296] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e09 - The Well-Tempered Clavier.mp4,
[2021-03-22T16:29:04.297] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e10 - The Bicameral Mind.mp4,
[2021-03-22T16:29:04.298] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e05 - Caravan.mp4,
[2021-03-22T16:29:04.299] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e06 - Its all in the game.mp4,
[2021-03-22T16:29:04.300] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e08 - Schijn bedriegt.mp4,
[2021-03-22T16:29:04.301] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e09 - De vrijgezellen.mp4,
[2021-03-22T16:29:04.301] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e10 - Onverwachte wending.mp4,
[2021-03-22T16:29:04.304] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e11 - Omkoopschandaal.mp4,
[2021-03-22T16:29:04.305] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e12 - Race tegen de klok.mp4,
[2021-03-22T16:29:04.305] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e13 - Moederziel en kinderleed.mp4,
[2021-03-22T16:29:04.306] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 2/Westworld - s02e02 - Reunion.mp4,
[2021-03-22T16:29:04.307] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 2/Westworld - s02e03 - Virtù e Fortuna.mp4,
[2021-03-22T16:29:04.307] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 2/Westworld - s02e05 - Akane No Mai.mp4,
[2021-03-22T16:29:04.311] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e01 - Nieuwbloed.mp4,
[2021-03-22T16:29:04.312] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 2/Westworld - s02e07 - Les Écorchés.mp4,
[2021-03-22T16:29:04.312] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e02 - Het drama.mp4,
[2021-03-22T16:29:04.317] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e03 - Oude bekende.mp4,
[2021-03-22T16:29:04.317] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e07 - Groot nieuws.mp4,
[2021-03-22T16:29:04.318] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e04 - Russische roulette.mp4,
[2021-03-22T16:29:04.318] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e05 - Vader.mp4,
[2021-03-22T16:29:04.319] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e06 - Uitweg.mp4,
[2021-03-22T16:29:04.319] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e07 - Geld.mp4,
[2021-03-22T16:29:04.321] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e08 - Creatief denkien.mp4,
[2021-03-22T16:29:04.322] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e09 - Ontvoerd.mp4,
[2021-03-22T16:29:04.322] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e10 - Tante Madonna.mp4,
[2021-03-22T16:29:04.323] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e11 - Voorzorgen.mp4,
[2021-03-22T16:29:04.324] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e12 - Oost west, thuis best.mp4,
[2021-03-22T16:29:04.325] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Voetbalvrouwen/Season 3/Voetbalvrouwen - s03e13 - De bedreigde bruid.mp4,
[2021-03-22T16:29:04.326] [INFO] Tdarr_Server - Bo6A1DCQc: File detected, adding to queue:/mnt/tvseries/Series/Westworld/Season 1/Westworld - s01e08 - Trace Decay.mp4,
[2021-03-22T16:29:15.607] [INFO] Tdarr_Server - Bo6A1DCQc: Sending files for scanning, library :Bo6A1DCQc,
[2021-03-22T16:29:18.023] [INFO] Tdarr_Server - nHkHQNXNXi Prep started,
[2021-03-22T16:29:18.288] [INFO] Tdarr_Server - [0.267s] nHkHQNXNXi Prep finished,
[2021-03-22T16:29:18.289] [INFO] Tdarr_Server - Scanner nHkHQNXNXi launched,
[2021-03-22T16:29:21.019] [INFO] Tdarr_Server - Online,
[2021-03-22T16:29:21.268] [INFO] Tdarr_Server - 18 files sending,
[2021-03-22T16:29:21.280] [INFO] Tdarr_Server - [0.012s] 18 files sent,
[2021-03-22T16:31:20.106] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e02 - Schuld en boete.mkv,
[2021-03-22T16:31:20.108] [INFO] Tdarr_Server - Bo6A1DCQc: file removed, removing:/mnt/tvseries/Series/Voetbalvrouwen/Season 2/Voetbalvrouwen - s02e04 - Het clublied.mkv,
[2021-03-22T16:32:23.966] [INFO] Tdarr_Server - Finished in 182.946s,
[2021-03-22T16:32:24.837] [INFO] Tdarr_Server - Scanner nHkHQNXNXi:Finished,
[2021-03-22T16:32:24.838] [WARN] Tdarr_Server - Exit approved. Closing process.,
[2021-03-22T16:32:25.902] [INFO] Tdarr_Server - nHkHQNXNXi File scanner exited,
[2021-03-22T16:40:35.415] [ERROR] Tdarr_Server - Error: Timeout: Did not receive an init message from worker after 10000ms. Make sure the worker calls expose().,
    at Timeout._onTimeout (/snapshot/tdarr_express_be/node_modules/threads/dist/master/spawn.js:35:53),
    at listOnTimeout (internal/timers.js:549:17),
    at processTimers (internal/timers.js:492:7),
[2021-03-22T16:40:35.422] [ERROR] Tdarr_Server - Error: Timeout: Did not receive an init message from worker after 10000ms. Make sure the worker calls expose().,
    at Timeout._onTimeout (/snapshot/tdarr_express_be/node_modules/threads/dist/master/spawn.js:35:53),
    at listOnTimeout (internal/timers.js:549:17),
    at processTimers (internal/timers.js:492:7),
```
I had the page/site open for most of the day (over 6 hours). I've only seen this happening when leaving the browser open for several hours. Restarting the Tdarr server solves the issue.

**Comment by HaveAGitGat** [comment]
2.00.08 which is in prelease (just built for all platforms) has the webui integrated within the server. The server auto restarts the webui every 10 mins or so which hopefully fixes this issue for Firefox.

**Comment by jcouch93** [comment]
> 
> 
> 2.00.08 which is in prelease (just built for all platforms) has the webui integrated within the server. The server auto restarts the webui every 10 mins or so which hopefully fixes this issue for Firefox.

Glad that there's a workaround on the way, but is there an idea of why or what is happening?  :)  Thanks!

**Comment by slim186** [comment]
I tried with Chrome.  After having a window open for approximately 5 days with no heavy loads (i.e. no rescans), the Chrome tab crashed with an out-of-memory error.  Looks like maybe a memory leak in the UI code?

**Comment by HaveAGitGat** [comment]
Thanks, earlier this week I found several memory issues on the server and in the browser which I've since fixed. In v<=2.00.08 there's a bug where the more you use the UI, the slower the UI and server become.

I've added some debug graphs for 2.00.09 which will make spotting memory issues easier in future.:
![image](https://user-images.githubusercontent.com/43864057/113395385-0bf84a00-939a-11eb-8f9f-bfb74424a708.png)

Will release this weekend.
